### ISSUE TITLE
Enhancement: defaultTimeOffset/defaultTimeRange full unit ranges.

### ISSUE THREAD

**Issue by Vynn** [alexarch21/history-explorer-card#153]
Right now the `defaultTimeRange` treats something like "1d" as an equivalent to 24h with the right-end of the x-axis being set at the present. Adding a units _should_ make the graph cover the period specified by the unit, or at least allow for the option to enable/disable this behavior.

Current Behavior: `defaultTimeRange: 24` shows a range starting with <currentTime> and showing the previous 24h of data stretching back from the present. The behavior is identical for `defaultTimeRange: 1d` and `defaultTimeRange: 24h`

Expected Behavior: `defaultTimeRange: 24` shows a range starting with <currentTime> and showing the previous 24h of data stretching back from the present. `defaultTimeRange: 24h` shows the current (incomplete) hour plus the 23h prior. `defaultTimeRange: 1d` should show the current day, beginning at midnight.

Another possibility implementation would be to default to showing full <x range> period when the range is set to a single unit: 1d, 1w, 1o, etc and defaulting to partial periods for multiple units: 24h, 7d, 4w. This could allow some flexibility without having to expose yet another configuration variable to the user.

**Comment by alexarch21** [comment]
I have to disagree with this. 24 hours or 1 day are just different units for the exact same timespan. They should not behave differently. Having multiple timeunits behave differently from single units, on the same parameter, is very obscure and non-discoverable behavior. 

If different behavior regarding time offsets is required, like snapping to absolute time instead of a relative offset, then this would have to be exposed by another parameter.

**Comment by Vynn** [comment]
It feels counterintuitive to me then that the behavior when scrolling through periods via the < [period unit] > arrows in the header follows the behavior I described as "expected". 

There are two models used at present to determine displayed ranges, and while the two behaviors are predictable, there's no ability to directly choose which is being used or apply it selectively.

**Comment by alexarch21** [comment]
> It feels counterintuitive to me then that the behavior when scrolling through periods via the < [period unit] > arrows in the header follows the behavior I described as "expected".

Not exactly sure what you mean by '< period unit >'. If you mean the range selector on the right side, that behaves exactly the same way as `defaultTimeRange`, it even calls the same function under the hood. If you mean the day selector on the left side, that one snaps to the next day boundary because it represents an absolute point in time (the date shown) rather than a time span. `defaultTimeRange` and `defaultTimeOffset` both refer to relative time spans.

**Comment by Vynn** [comment]
Yes, I guess it would be the “day selector”, although I expected it would work to jump along whatever whatever finite increment of time was set for the range. 

So then the enhancement would be maybe better requested as a way to declare and display an absolute range. It’s something I’ve been trying to do on initial display, but can’t do it with the available options unless I’ve missed something in the documentation (which is excellent).

**Comment by Vynn** [comment]
Alternatively, if the Day Selector followed whatever range was in the Range Selector that would work too. 

The reason for the enhancement request is I'm not able to display the card, page back and forth through a fixed range of time (in this case, 1 day), and then return to the initial view as loaded without closing or refreshing the card.

Makes it much harder to do quick comparisons.

**Comment by alexarch21** [comment]
Have you looked into issue #151 ? The user there uses custom button card to programmatically set `defaultTimeRange` and `defaultTimeOffset`. You could use this to dynamically clamp the visible range to day boundaries.

> Alternatively, if the Day Selector followed whatever range was in the Range Selector that would work too.

This has been on the backlog for a while and I'm glad you brought it up again. If the range is set to, like, a month, then the day selector should become a month selector. Possibly with some other granularity inbetween (week ?).

> The reason for the enhancement request is I'm not able to display the card, page back and forth through a fixed range of time (in this case, 1 day), and then return to the initial view as loaded without closing or refreshing the card.

You can click on the day selector, it will bring you back to the initial view with your `defaultTimeOffset`.

**Comment by alexarch21** [comment]
> Alternatively, if the Day Selector followed whatever range was in the Range Selector that would work too.

As of V1.0.45 the date selector previous and next buttons will scroll by a month at a time at long time ranges.
